### PR TITLE
[5.2] Support IronMQ

### PR DESCRIPTION
**Pull request by alex-litvak**

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
As I explained on your last PR, we intentionally removed support for this in 5.2.

File: composer.json [hunk 1]
```diff
@@ -72,6 +72,7 @@
     },
     "require-dev": {
         "aws/aws-sdk-php": "~3.0",
+        "iron-io/iron_mq": "~4.0.1",
```
Comment (GrahamCampbell) [inline composer.json]: This needs to go in the suggests block in BOTH composer.json files please.

File: src/Illuminate/Queue/Connectors/IronConnector.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Illuminate\Queue\Connectors;
+
+use Illuminate\Queue\IronQueue;
+use Illuminate\Contracts\Encryption\Encrypter as EncrypterContract;
+use Illuminate\Http\Request;
+use IronMQ\IronMQ;
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/Connectors/IronConnector.php]: Please sort by length.

File: src/Illuminate/Queue/Connectors/IronConnector.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        $this->crypt = $crypt;
+        $this->request = $request;
+    }
+
+    /**
+     * Establish a queue connection.
+     *
+     * @param  array  $config
+     * @param  array  $config
+     *
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/Connectors/IronConnector.php]: please but no extra line here in all cases

File: composer.json (hunk 1 above)
Comment (alex-litvak) [inline composer.json]: @GrahamCampbell Thanks, I will add it to suggests block but what do you mean by "BOTH" composer.json files?

Comment (GrahamCampbell) [inline composer.json]: There is a composer.json file in src/Illuminate/Queue.

**Comment by alex-litvak** [comment]
@GrahamCampbell Done. I should also add artisan command for IronMQ (push_queues). Can I do it in the next PR?

**Comment by taylorotwell** [comment]
Can we send this over to the https://github.com/LaravelCollective/iron-queue repository where this is maintained. That organization exists to take some load off of me and we can probably even add you as a collaborator on that repository. /cc @adamgoose

**Pull request closed without merging**